### PR TITLE
feat(scrubbing): Scrub `span.data.http.query` with default scrubbers

### PR DESCRIPTION
**Pull request by iker-barriocanal**
Ref: https://github.com/getsentry/relay/issues/1855

`span.data.http.query` may contain sensitive data, which is currently not scrubbed. This PR makes relay to scrub that field with default data scrubbers.

### CODE REVIEW

**Review by cleptric — APPROVED** [review]
LGTM, we may add an additional test case that checks a more complex query, like `api_token=123&process_id=123` -> `api_token=[Filtered]&process_id=123`

File: relay-general/src/protocol/types.rs [hunk 1]
```diff
@@ -1117,6 +1117,26 @@ impl schemars::JsonSchema for Timestamp {
...
+
+    #[metastructure(additional_properties, retain = "true", pii = "maybe")]
+    pub other: Object<Value>,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, Empty, FromValue, IntoValue, ProcessValue)]
+#[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
+pub struct HttpElement {
+    #[metastructure(pii = "true")]
+    pub query: Annotated<String>,
```
Comment (jjbayer) [inline relay-general/src/protocol/types.rs]: `span.data` is [documented as free-form](https://develop.sentry.dev/sdk/event-payloads/span/), so when a user sends `{"http": {"query": {"not": "a string"}}}`, we should still accept it.

@iker-barriocanal could you check if the scrubbing still works when you change `String` to `Value` here?

**Review by jjbayer — CHANGES_REQUESTED** [review]

File: relay-general/src/protocol/types.rs (hunk 1 above)
Comment (iker-barriocanal) [reply relay-general/src/protocol/types.rs]: Great catch! This is fixed in 03af9eb46153a99ecada9db886da4c6dab5ffbaf, and continues to work as expected.

File: relay-general/src/protocol/span.rs [hunk 2]
```diff
@@ -70,7 +74,12 @@ mod tests {
   "op": "operation",
   "span_id": "fa90fdead5f74052",
   "trace_id": "4c79f60c11214eb38604f4ae0781bfb2",
-  "status": "ok"
+  "status": "ok",
+  "data": {
+    "http": {
+      "query": "is_sample_query=true"
+    }
+  }
```
Comment (jjbayer) [inline relay-general/src/protocol/span.rs:89]: Should we add some random extra fields in `data` and `data.http` here to validate that arbitrary data is still accepted?

**Review by jjbayer — APPROVED** [review]

**Pull request merged**